clients/roscpp/src/libros/callback_queue.cpp

```diff
@@ -382,14 +393,6 @@ CallbackQueue::CallOneResult CallbackQueue::callOneCB(TLS* tls)
         result = cb->call();
       }
     }
-    catch (std::exception&)
-    {
-      // ensure that thread id gets restored, even in case of an exception
-      tls->calling_in_this_thread = last_calling;
-      throw;
```

Comment:
Nvm, I didn't think about it before. While it passes along the same exception it will change the reported line in e.g. gdb which is clearly undesired.